pkg/new-ui/v1beta1/frontend/src/app/pages/experiment-creation/hyper-parameters/add-modal/add-modal.component.ts

```diff
@@ -0,0 +1,70 @@
+import { Component, OnInit, OnDestroy, Inject } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { FormGroup, FormControl, Validators } from '@angular/forms';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-add-param-modal',
+  templateUrl: './add-modal.component.html',
+  styleUrls: ['./add-modal.component.scss'],
+})
+export class AddParamModalComponent implements OnInit, OnDestroy {
+  formGroup: FormGroup;
+
+  subs = new Subscription();
+
+  get isList(): boolean {
+    const tp = this.formGroup.get('type').value;
+    return tp === 'discrete' || tp === 'categorical';
+  }
+
+  get inputStep(): number {
+    const tp = this.formGroup.get('type').value;
+    if (tp === 'int') {
+      return 1;
+    }
+
+    return 0.01;
+  }
+
+  constructor(
+    private dialog: MatDialogRef<AddParamModalComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: FormGroup,
+  ) {
+    this.formGroup = data;
+  }
+
+  ngOnInit() {
+    this.subs.add(
+      this.formGroup.get('type').valueChanges.subscribe(type => {
+        if (this.isList) {
+          this.formGroup.removeControl('value');
+          this.formGroup.addControl('value', new FormControl('yo'));
+          return;
+        }
+
+        this.formGroup.removeControl('value');
+        this.formGroup.addControl(
+          'value',
+          new FormGroup({
+            min: new FormControl(1, Validators.required),
+            max: new FormControl(64, Validators.required),
+            step: new FormControl(1, Validators.required),
```

Comment:
Step parameter is required only for the Grid algorithm.
I think we can avoid this requirements.
Suggestion CR should handle errors/validation.

Reply:
ACK, I'll make this input optional then.
Out of curiosity, if this is omitted then what value will be used? Or is the suggestions controller only going to use the min and the max value?

Reply:
It depends on the algorithm.
For example, skopt with BO uses log uniform distribution where user has to set only max and min value https://github.com/kubeflow/katib/blob/master/pkg/suggestion/v1beta1/skopt/base_service.py#L43-L44

Reply:
Thanks for the info!

Then I'll:
* not make this field **required** in the form. By default it will be empty 
* the text in the UI, on top of the input, will say `Step (optional)` [previously it didn't say optional]
* make the UI completely omit this field in the CR if the value is the empty string

@andreyvelich does this sound good to you?

Reply:
I took the liberty and implemented the above. If you'd like something different let me know and I'll modify

Reply:
It looks great, thank you @kimwnasptd!